extensions/dub-link-shortener/src/shorten-link.tsx

```diff
@@ -1,99 +1,140 @@
+              onAction: () => open("raycast://extensions/quuu/dub-link-shortener/search-links"),
+            },
+          });
+        })
+        .catch(async (err) => {
+          captureException(err);
+          const toast = await showFailureToast(err, { title: "❗ Failed to create link" });
+          toast.primaryAction = {
+            title: "Retry",
+            onAction: () => {
+              push(<ShortenLinkForm retryValues={vals} args={args} />);
+              toast.hide();
+            },
+          };
+        })
+        .finally(pop);
+    },
+    validation: {
+      url: FormValidation.Required,
+      domain: FormValidation.Required,
+      workspaceId: FormValidation.Required,
+    },
+    initialValues: retryValues ?? { domain: "dub.sh", workspaceId, url: isEmpty(url) ? originalLink : url, key },
+  });
 
   return (
     <Form
-      isLoading={isLoading}
+      searchBarAccessory={<Form.LinkAccessory text="Dashboard" target="https://app.dub.co" />}
+      isLoading={isLoading || isLoadingLink}
       actions={
         <ActionPanel>
           <Action.SubmitForm title="Shorten Link" icon={Icon.Link} onSubmit={handleSubmit} />
         </ActionPanel>
       }
     >
-      <Form.Description text="Your shortened link will be copied to your clipboard." />
-      <Form.TextField
-        id="url"
-        title="Original URL"
-        placeholder="https://dub.co"
-        value={originalLink}
-        onChange={(newValue) => setOriginalLink(newValue)}
-      />
-      <Form.TextField
-        id="shortUrl"
-        title="URL Key"
-        placeholder="(Optional)"
-        value={urlKey}
-        onChange={(urlKey) => setUrlKey(urlKey)}
-      />
-      {lastShortenedUrl !== "" && originalLink === "" && (
-        <>
-          <Form.Description text={`Shortened link (copied to clipboard):`} />
-          <Form.Description text={`${lastShortenedUrl}`} />
-        </>
-      )}
+      <Form.Description title={"❗"} text="Your shortened link will be copied to your clipboard." />
+      <Form.Dropdown {...itemProps.workspaceId} title="Workspace">
+        {workspaces.map((w) => (
+          <Form.Dropdown.Item
+            key={w.id}
+            value={w.id}
+            title={w.name}
+            icon={{ source: w.logo ?? "command-icon.png", mask: Image.Mask.Circle }}
+          />
+        ))}
+      </Form.Dropdown>
+      <Form.Separator />
+      <Form.TextField {...itemProps.url} placeholder="https://dub.co" title="Original URL" />
+      <Form.TextField {...itemProps.key} placeholder="(Optional)" title="URL Key" />
+      <Form.TagPicker {...itemProps.tagIds} title="Tags">
+        {workspaces
+          .find((w) => w.id === values.workspaceId)
+          ?.tags?.map((t) => (
+            <Form.TagPicker.Item
+              key={t.id}
+              value={t.id}
+              title={t.name}
+              icon={{ source: Icon.Tag, tintColor: Color.Orange }}
+            />
+          ))}
+      </Form.TagPicker>
```

Comment:
this only allows selecting tags that were already created elsewhere -- this requires adding the tag outside of raycast for now?

Reply:
Yeah this search is limited to the already existing tags in the workspace. Raycast tag-picker does not allow for dynamic tag creations, so we will need a new form to add tags to a workspace so that it can be used in following link creations.